src/System.Private.CoreLib/shared/System/DefaultBinder.cs

```diff
@@ -68,7 +69,7 @@ partial class DefaultBinder : Binder
                     // Named parameters, reorder the mapping.  If CreateParamOrder fails, it means that the method
                     // doesn't have a name that matchs one of the named parameters so we don't consider it any further.
                     if (!CreateParamOrder(paramOrder[i], par, names))
-                        candidates[i] = null;
+                        candidates[i] = null!;
```

Comment:
should `MethodBase[] candidates` be `MethodBase?[] candidates` instead?

Reply:
it can be, but there is lot of references to `candidates[i]` without null check. This is local variable, so doesn't have much effect which ever i choose (`candidates?[i]` or `candidates[i]`). So I decide if make it nullable or not based on how many `!` i need put if choose any of them, so in this case if I make it `candidates?[i]` i will need to put `!` in much more rows